riscv/interactive.cc

```diff
@@ -58,6 +75,63 @@ static std::string readline(int fd)
   return s;
 }
 
+#ifdef HAVE_BOOST
+// read input command string
+string sim_t::rin(streambuf *bout_ptr) {
+   string s;
+   if (acceptor_ptr) { // if we are listening, get commands from socket
+      try
+      {
+         socket_ptr = new tcp::socket(*io_service_ptr);
+         acceptor_ptr->accept(*socket_ptr); // wait for someone to open connection
+         boost::asio::streambuf buf;
+         boost::asio::read_until(*socket_ptr, buf, "\n"); // wait for command
+         s = buffer_cast<const char*>(buf.data());
+         erase_all(s, "\r");  // get rid off any cr and lf
+         erase_all(s, "\n");
+         // The socket client is a web server and it appends the IP of the computer
+         // that sent the command from its web browser.
+
+         // For now, erase the IP if it is there.
+         regex re(" ((25[0-5]|2[0-4][0-9]|[01]?[0-9]?[0-9])\\.){3}"
+                  "(25[0-5]|2[0-4][0-9]|[01]?[0-9]?[0-9])$");
+         s = regex_replace(s, re, (string)"");
+
+         // TODO: check the IP against the IP used to upload RISC-V source files
```

Comment:
No.
When the web server interface is used to input debug commands, it appends the IP from where the web server has been accessed, to the command line. When the assembly file is uploaded, it is also marked with the IP from where it was uploaded, so it is possible to check if the upload IP and the debug input IP are the same.
At least this semester, I do not intend to implement this check.
However, the IP which has been appended needs to be removed from the debug input command line.
When a local GUI application is used instead of the web server interface, the GUI does not append any IP address. In this case, regex_replace will not find any IP address and it will leave the command line unchanged.

Reply:
What/where is this "web server interface"? Is the message format (which includes the IPs) documented anywhere?

Reply:
A paper, which documents the message protocol, has been submitted to a conference, however the program comittee decision did not arrive yet.